mainwindow.cpp

```diff
@@ -142,9 +195,105 @@ void MainWindow::toggleOutputSplitter()
-        return true;
+    if (target == ui->mainOutput->viewport())
+    {
+        if (event->type() == QEvent::Resize)
+        {
+            // re position search widget when main output inner size changes
+            // this takes into account existence of vertical scrollbar
+            QResizeEvent *resizeEvent = static_cast<QResizeEvent*>(event);
+            search_widget->move(resizeEvent->size().width() - search_widget->width(), 0);
+        }
+        else if (event->type() == QEvent::Wheel)
+        {
+            // allow mouse wheel usage only in 'browsing mode'
+            if (!ui->bottomOutput->isVisible())
+                return true;
+        }
+    }
+    else
+    {
+        if (event->type() == QEvent::KeyPress)
+        {
+            QKeyEvent *keyEvent = static_cast<QKeyEvent*>(event);
+            if (ui->searchButton->isChecked())
+            {
+                // hide search widget on Escape key press
+                if (keyEvent->key() == Qt::Key_Escape)
+                    ui->searchButton->toggle();
+            }
+            else
+            {
+                // show search widget on Ctrl-F
+                if (keyEvent->key() == Qt::Key_F && keyEvent->modifiers() == Qt::ControlModifier)
+                    ui->searchButton->toggle();
+            }
+            if (target == search_input)
+            {
+                if (keyEvent->key() == Qt::Key_Return)
+                {
+                    if (keyEvent->modifiers() == Qt::ShiftModifier)
+                        search_prev_button->click();
+                    else
+                        search_next_button->click();
+                }
+            }
+
+        }
+    }
 
     // base class behaviour
     return QMainWindow::eventFilter(target, event);
 }
+
+void MainWindow::showSearchWidget(bool show)
+{
+    // record which widget had focus before showing search widget, in order
+    // to return focus to it when search widget is hidden
+    static QWidget *prevFocus = 0;
+
+    QPropertyAnimation *animation = new QPropertyAnimation(search_widget, "pos");
+    animation->setDuration(150);
+
+    // place search widget at the rightmost position of mainoutput viewport
+    QPoint showed_pos(ui->mainOutput->viewport()->width() - search_widget->width(), 0);
+    QPoint hidden_pos(ui->mainOutput->viewport()->width() - search_widget->width(), -search_widget->height());
+    animation->setStartValue(show ? hidden_pos : showed_pos);
+    animation->setEndValue(show ? showed_pos : hidden_pos);
+
+    if (show)
+    {
+        prevFocus = QApplication::focusWidget();
+        search_widget->setVisible(show);
+        search_input->setFocus();
+    }
+    else
+    {
+        connect(animation, &QPropertyAnimation::destroyed, search_widget, &QWidget::hide);
+        prevFocus->setFocus();
+    }
+    animation->start(QAbstractAnimation::DeleteWhenStopped);
+}
+
+void MainWindow::handleCursosPosChanged(int pos)
+{
+    // only in browsing mode
```

Comment:
perché soltanto in "browsing mode"? E' vero che se arriva altro testo la posizione del cursore verrà automaticamente settata alla fine del documento, ma penso sia comunque utile (o perlomeno più intuitivo e non certo dannoso) che il documento possa essere scrollato in automatico per "seguire" il cursor della ricerca.